lib/internal/bootstrap/node.js

```diff
@@ -574,7 +574,7 @@
       // if node-report is enabled, call into its handler to see
       // whether it is interested in handling the situation.
       // Ignore if the error is scopped inside a domain.
-      if (er.domain == null) {
+      if (er === undefined || er === null || er.domain === null) {
```

Comment:
this is the change that causes `test/node-report/test-exception.js` to fail. I am unsure why this is required; can you please explain?

we want to enter the block only if the control flow is not under the control / scope of a domain, and hence my `if (er.domain == null)` check.

we are already in uncaught `_fatalException` handler that means we definitely have an error condition, so `er` object is going to be non-null when we reach here.

reverting this piece of change passes the test.

Reply:
It does not seem to be the case that er is going to be non-null, or not undefined (in that case the change should have no effect anyway) The problem was that a number of tests failed because er was undefined and we'd get an exception on er.domain.  

Running the test suite with that change backed out results in X failures in which the issue being reported is that er is undefined.  For example:

```bash
Command: out/Release/node /home/mhdawson/node-report/node/test/message/throw_null.js
[01:10|%  94|+ 2309|-   4]: release throw_undefined.jslength differs.
expect=4
actual=5
patterns:
pattern = ^.*test.*message.*throw\_undefined\.js\:.*$
pattern = ^throw\ undefined\;$
pattern = ^\^$
pattern = ^undefined$
outlines:
outline = internal/bootstrap/node.js:578
outline =       if (er.domain == null) {
outline =              ^
outline = TypeError: Cannot read property 'domain' of undefined
outline =     at process._fatalException (internal/bootstrap/node.js:578:14)
=== release throw_undefined.js ===
Path: message/throw_undefined.js
internal/bootstrap/node.js:578
      if (er.domain == null) {
             ^

TypeError: Cannot read property 'domain' of undefined
    at process._fatalException (internal/bootstrap/node.js:578:14)
Command: out/Release/node /home/mhdawson/node-report/node/test/message/throw_undefined.js

```

Reply:
My guess as to why 'test/node-report/test-exception.js' is that without the check we end up with a "fatal" exception and that for some reason a node-report is only being generated if there is a fatal exception versus an uncaught exception

Reply:
Ok I think I figured out the issue.  It was with this part of the change:

`er.domain === null`

That limited the check to stricly null intead of null or undefined I think.  Changed it back to a double equals and left in the other checks.  Now checking to see if all tests pass.

Reply:
Yes, that was the problem. Pushed another change which moves it back to == and now all of the tests pass.

Reply:
thanks @mhdawson - that looks good to me. So squashed all the commits so far.

btw, curious to know which platform you ran the tests? I never saw this error (`Cannot read property 'domain' of undefined`) was the reason why the code was developed as it was.